include/deal.II/matrix_free/operators.h

```diff
@@ -246,7 +248,7 @@ namespace MatrixFreeOperators
      * order to ensure that the cell loops will be able to access the ghost
      * indices with the correct local indices.
      */
-    void adjust_ghost_range_if_necessary(const LinearAlgebra::distributed::Vector<Number> &vec) const;
+    void adjust_ghost_range_if_necessary(const VectorType &vec) const;
```

Comment:
yes, i think we can move it out. Either a stand-alone function templated with `VectorType` which receives `MPI::Partitioner` or make it a part of each parallel vector. Probably the former is more flexible.